### ISSUE TITLE
WGAN-GP batch size

### ISSUE THREAD

**Issue by ibiscp** [eriklindernoren/Keras-GAN#167]
I would like to know how to train WGAN-GP with variable batch sizes. I couldn't find a way to set the batch size in the function below dynamically. This is needed because my last batch is not complete.

```
class RandomWeightedAverage(Add):
    """Provides a (random) weighted average between real and generated image samples"""
    def _merge_function(self, inputs):
        alpha = K.random_uniform((32, 1, 1, 1))
        return (alpha * inputs[0]) + ((1 - alpha) * inputs[1])
```

**Comment by Rabia-Raja** [comment]
Hi can you explain me please why there is no accuracy value in wgan and wgan-gp code?

**Comment by saundersp** [comment]
Here's how i did it.

```python
class RandomWeightedAverage(Add):
    # Provides a (random) weighted average between real and generated image samples
    def _merge_function(self, inputs):
        input1, input2 = inputs
        alpha = K.random_uniform((K.shape(input1)[0], 1, 1, 1))
        return (alpha * input1) + ((1 - alpha) * input2)
```
